sdk/appconfiguration/app-configuration/samples-dev/featureFlag.ts

```diff
@@ -22,124 +21,146 @@ dotenv.config();
-    // setting is a `FeatureFlag`
-    const conditions = getResponse.conditions;
-
-    // the client filters are the real meat of the FeatureFlag.
-    //
-    for (const clientFilter of conditions.clientFilters) {
-      if (isFeatureFlagClientFilter("targeting", clientFilter)) {
-        // some of the fields:
-        // clientFilter.parameters.audience
-        // clientFilter.parameters.audience.groups[0].name
-        // clientFilter.parameters.audience.groups[0].rolloutPercentage
-        // clientFilter.parameters.audience.users[0]         // string
-        // clientFilter.parameters.audience.defaultRolloutPercentage
-        console.log(
-          `  targeting feature flag client filter => name: ${clientFilter.name}, defaultRolloutPercentage: ${clientFilter.parameters.audience.defaultRolloutPercentage}`
-        );
-        clientFilter.parameters.audience.defaultRolloutPercentage = 85;
-      } else if (isFeatureFlagClientFilter("timeWindow", clientFilter)) {
-        // clientFilter.parameters.end;
-        // clientFilter.parameters.start;
-        console.log(
-          `  timeWindow feature flag client filter => name: ${clientFilter.name}, start time: ${clientFilter.parameters.start}`
-        );
-        clientFilter.parameters.start = "Wed, 01 June 2021 13:59:59 GMT";
-      } else if (isFeatureFlagClientFilter("percentage", clientFilter)) {
-        console.log(
-          `  percentage feature flag client filter => name: ${clientFilter.name}, value: ${clientFilter.parameters.value}`
-        );
-        clientFilter.parameters.value = 56;
-      } else {
-        console.log(
-          `  name of the custom feature flag client filter => name : ${clientFilter.name}`
-        );
-        clientFilter.name = "FilterY";
-      }
+  // You can use the `isFeatureFlag` global method to check if the content type is featureFlagContentType ("application/vnd.microsoft.appconfig.ff+json;charset=utf-8")
+  const newFeatureFlag = parseFeatureFlag(getResponse); // Converts the configurationsetting into featureflag
+  // Modify the props
+  for (const clientFilter of newFeatureFlag.value.conditions.clientFilters) {
+    if (clientFilter.name === "Microsoft.Targeting") {
+      // some of the fields:
+      // clientFilter.parameters.audience
+      // clientFilter.parameters.audience.groups[0].name
+      // clientFilter.parameters.audience.groups[0].rolloutPercentage
+      // clientFilter.parameters.audience.users[0]         // string
+      // clientFilter.parameters.audience.defaultRolloutPercentage
+      console.log(
+        `  targeting feature flag client filter => name: ${
+          clientFilter.name
+        }, audience: ${JSON.stringify(clientFilter.parameters!.audience, null, 2)}`
+      );
+      clientFilter.parameters = {
+        ...clientFilter.parameters,
+        audience: {
+          ...(clientFilter.parameters!.audience as {
```

Comment:
this case feels kinda gross, is there a nicer way we can accomplish this?

Reply:
Thanks for pointing out!
I also was using the older "lowercase" convention.
Updated this weirdness and the casing as well, please let me know if that looks good!